src/test/scala/helpers/TestHelper.scala

```diff
@@ -35,18 +35,20 @@ object TestHelper {
   }
 
   def getDbSettings() =
-      DbSettings(getDriverFromEnv,      // driver
-          System.getenv("DBCONNSTRING"),  // nrl
-          System.getenv("DBUSER"),        // user
-          System.getenv("DBPASSWORD"),    // password
-          System.getenv("DBNAME"),        // dbname
-          System.getenv("DBHOST"),
-          System.getenv("DBPORT"),
-          System.getenv("GPHOST"),
-          System.getenv("GPPATH"),
-          System.getenv("GPPORT"),
-          getGPLOADEnv(),
-          getIsIncrementalEnv())
+      DbSettings(
```

Comment:
I'm against using Map.  I initially thought we should directly use the Config object instead of creating an array of Settings classes.  But it seems Config is awkward to use (throws exception for null) and since we already have this typed intermediate representation it's good to maintain it cleanly.
Using case classes with named argument syntax and default values eliminates the need for impacting all constructors.  And right, using `.copy()` on top of that further reduces the impact.  This PR is actually all about these two to make it easier to further refactor things.